src/hetzner_dns.rs

```diff
@@ -0,0 +1,69 @@
+use async_trait::async_trait;
+use reqwest::Client;
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+
+use crate::{
+    config::HetznerConfig,
+    consul,
+    dns_trait::{DnsProviderTrait, DnsRecord},
+};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct RecordsWrapper {
+    records: Vec<DnsRecord>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct RecordResponse {
+    record: DnsRecord,
+}
+
+pub struct HetznerDns {
+    pub config: HetznerConfig,
+}
+
+#[async_trait]
+impl DnsProviderTrait for HetznerDns {
+    /// Create a DNS record based on the Consul service tags
+    async fn create_dns_record<'a>(
+        &self,
+        dns_record: &'a consul::DnsRecord,
+    ) -> Result<String, anyhow::Error> {
+        let new_record = json!({
+            "zone_id": self.config.dns_zone_id,
+            "type": dns_record.type_,
+            "name": dns_record.hostname,
+            "value": dns_record.value,
+            "ttl": dns_record.ttl
+        });
+
+        let client = Client::new();
```

Comment:
We really want to store a `Client` inside the `HetznerDns` struct so that we don't have to set up a new TCP connection for every API request.

Reply:
fixed